### PR TITLE
Remove components from root module

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

**Review by dmitryax — APPROVED** [review]

**Review by mx-psi — COMMENTED** [review]
Do we need a root go module at all after this PR? Can we deprecate it if not?

**Review by codeboten — APPROVED** [review]
🎉 

Please rebase

**Review by atoulme — APPROVED** [review]
Please rebase?

**Comment by atoulme** [comment]
It won't be very good, but I'm trying to rebase this PR in #19456

**Comment by atoulme** [comment]
My rebased PR passes: #19456

File: cmd/configschema/go.mod [hunk 1]
```diff
@@ -5,11 +5,171 @@ go 1.18
 require (
 	github.com/fatih/structtag v1.2.0
 	github.com/google/uuid v1.3.0
-	github.com/open-telemetry/opentelemetry-collector-contrib v0.70.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/alibabacloudlogserviceexporter v0.70.0
```
Comment (MovieStoreGuy) [inline cmd/configschema/go.mod:8]: These versions are using `v0.70.0`, it should be `v0.73.0` right?

**Review by MovieStoreGuy — COMMENTED** [review]
This PR seems a little off for me, not sure what is the go here.

File: cmd/configschema/go.mod (hunk 1 above)
Comment (atoulme) [reply cmd/configschema/go.mod:8]: this PR is really old now, see #19456 for a new take

**Pull request closed without merging**